src/ProtocolV3TestBase.sol

```diff
@@ -184,68 +225,202 @@ contract ProtocolV3TestBase is CommonTestBase {
     IPoolAddressesProvider addressesProvider = IPoolAddressesProvider(pool.ADDRESSES_PROVIDER());
     IAaveOracle oracle = IAaveOracle(addressesProvider.getPriceOracle());
     uint256 latestAnswer = oracle.getAssetPrice(config.underlying);
+
     return (dollarValue * 10 ** (8 + config.decimals)) / latestAnswer;
   }
 
-  function _e2eTestBorrowRepay(
+  function _getMaxBorrowAmount(
+    IPool pool,
+    ReserveConfig memory collateralConfig,
+    ReserveConfig memory borrowConfig,
+    uint256 collateralAmount
+  ) internal view returns (uint256) {
+    IPoolAddressesProvider addressesProvider = IPoolAddressesProvider(pool.ADDRESSES_PROVIDER());
+    IAaveOracle oracle = IAaveOracle(addressesProvider.getPriceOracle());
+    return collateralAmount
+      * oracle.getAssetPrice(collateralConfig.underlying)
+      * collateralConfig.ltv
+      * (10 ** borrowConfig.decimals)
+      / oracle.getAssetPrice(borrowConfig.underlying)
+      / (10 ** collateralConfig.decimals)
+      / 10_000;
```

Comment:
This is a bps value so I usually write it like `100_00` to signify 100%.

Reply:
Updated